File: BackEnd/Service/Models/CandidateJoinEventModel.cs
namespace Service.Models
{
    public class CandidateJoinEventModel
    {
        public Guid CandidateJoinEventId { get; set; }

        public Guid CandidateId { get; set; }

        public Guid EventId { get; set; }

        public DateTime? DateJoin { get; set; } = DateTime.Now;
        public int JoinEventCount { get; set; }

#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
        public EventModel Event { get; set; }
#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
        public CandidateModel Candidate { get; set; }
#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
    }
}

File: FRONTEND/src/app/data/candidateJoinEvent/candidate-join-event.model.ts
import { Candidate } from "../candidate/candidate.model";
import { Event } from "../event/event.model";

export class CandidateJoinEvent {
	candidateJoinEventId?: string;
	candidateid?: string;
	eventId?: string;
	dateJoin?: Date;
	event?: Event;
	candidate?: Candidate;
}
